src/internal/operations/backup_integration_test.go

```diff
@@ -1087,6 +1099,371 @@ func (suite *BackupOpIntegrationSuite) TestBackup_Run_oneDrive() {
 	runAndCheckBackup(t, ctx, &bo, mb)
 }
 
+func mustGetDefaultDriveID(
+	t *testing.T,
+	ctx context.Context, //revive:disable-line:context-as-argument
+	service graph.Servicer,
+	userID string,
+) string {
+	d, err := service.Client().UsersById(userID).Drive().Get(ctx, nil)
+	if err != nil {
+		err = graph.Wrap(
+			ctx,
+			err,
+			"failed to retrieve default user drive").
+			With("user", userID)
+	}
+
+	require.NoError(t, err)
+
+	id, ok := ptr.ValOK(d.GetId())
+	require.True(t, ok, "drive ID not set")
+	require.NotEmpty(t, id)
+
+	return id
+}
+
+// TestBackup_Run ensures that Integration Testing works for OneDrive
+func (suite *BackupOpIntegrationSuite) TestBackup_Run_oneDriveIncrementals() {
+	// TODO: Enable once we have https://github.com/alcionai/corso/pull/2642
+	suite.T().Skip("Enable once OneDrive incrementals is available")
+
+	ctx, flush := tester.NewContext()
+	defer flush()
+
+	var (
+		t    = suite.T()
+		acct = tester.NewM365Account(t)
+		ffs  = control.Toggles{}
+		mb   = evmock.NewBus()
+
+		// `now` has to be formatted with SimpleDateTimeOneDrive as
+		// some onedrive cannot have `:` in file/folder names
+		now = common.FormatNow(common.SimpleTimeTesting)
+
+		owners = []string{suite.user}
+
+		categories = map[path.CategoryType][]string{
+			path.FilesCategory: {graph.DeltaURLsFileName, graph.PreviousPathFileName},
+		}
+		container1 = fmt.Sprintf("%s%d_%s", incrementalsDestContainerPrefix, 1, now)
+		container2 = fmt.Sprintf("%s%d_%s", incrementalsDestContainerPrefix, 2, now)
+		container3 = fmt.Sprintf("%s%d_%s", incrementalsDestContainerPrefix, 3, now)
+
+		genDests = []string{container1, container2}
+	)
+
+	m365, err := acct.M365Config()
+	require.NoError(t, err)
+
+	gc, err := connector.NewGraphConnector(
+		ctx,
+		graph.HTTPClient(graph.NoTimeout()),
+		acct,
+		connector.Users,
+		fault.New(true))
+	require.NoError(t, err)
+
+	driveID := mustGetDefaultDriveID(t, ctx, gc.Service, suite.user)
+
+	fileDBF := func(id, timeStamp, subject, body string) []byte {
+		return []byte(id + subject)
+	}
+
+	// Populate initial test data.
+	// Generate 2 new folders with two items each. Only the first two
+	// folders will be part of the initial backup and
+	// incrementals. The third folder will be introduced partway
+	// through the changes. This should be enough to cover most delta
+	// actions.
+	for _, destName := range genDests {
+		generateContainerOfItems(
+			t,
+			ctx,
+			gc,
+			path.OneDriveService,
+			acct,
+			path.FilesCategory,
+			selectors.NewOneDriveRestore(owners).Selector,
+			m365.AzureTenantID, suite.user, driveID, destName,
+			2,
+			// Use an old backup version so we don't need metadata files.
```

Comment:
Do we plan on having a matrix of tests for the different backup versions?

Reply:
I think we can skip the matrix of tests here. Really the matrix tests that _restore_ is backwards compatible as we'll always make backups with the latest version

As an aside, the matrix of different versions tests should really be moved to a nightly regression test job because they're starting to take a non-trivial amount of time to run (hazards of having a test framework that is too easy to use... 😅)